fuse_examples/classification/multimodality/mg_dataset.py

```diff
@@ -0,0 +1,432 @@
+import pandas as pd
+import os
+from typing import Callable, Optional
+from typing import Tuple
+
+from MedicalAnalyticsCore.DatabaseUtils.selected_studies_queries import get_annotations_and_findings
+from MedicalAnalyticsCore.DatabaseUtils.tableResolver import TableResolver
+from MedicalAnalyticsCore.DatabaseUtils.connection import create_homer_engine, Connection
+from MedicalAnalyticsCore.DatabaseUtils import tableNames
+from MedicalAnalyticsCore.DatabaseUtils import db_utils as db
+
+
+# from autogluon.tabular import TabularPredictor
+from fuse_examples.classification.multimodality.dataset import IMAGING_TABULAR_dataset
+from fuse.data.dataset.dataset_default import FuseDatasetDefault
+
+from fuse_examples.classification.MG_CMMD.input_processor import FuseMGInputProcessor
+from fuse.data.processor.processor_dataframe import FuseProcessorDataFrame
+
+
+from typing import Dict, List
+import torch
+from fuse.utils.utils_hierarchical_dict import FuseUtilsHierarchicalDict
+
+class PostProcessing:
+    def __init__(self, continuous_tabular_features_lst: List, categorical_tabular_features_lst: List, label_lst: List,
+                 imaging_features_lst: List, non_imaging_features_lst: List, use_imaging: bool, use_non_imaging: bool):
+        self.continuous_tabular_features_lst = continuous_tabular_features_lst
+        self.categorical_tabular_features_lst = categorical_tabular_features_lst
+        self.label_lst = label_lst
+        self.imaging_features_lst = imaging_features_lst
+        self.non_imaging_features_lst = non_imaging_features_lst
+        self.use_imaging = use_imaging
+        self.use_non_imaging = use_non_imaging
+
+    def __call__(self, batch_dict: Dict) -> Dict:
+        if not self.use_imaging and not self.use_non_imaging:
+            raise ValueError('No features are in use')
+        mask_list = self.use_imaging * self.imaging_features_lst + self.use_non_imaging * self.non_imaging_features_lst
+        mask_continuous = torch.zeros(len( self.continuous_tabular_features_lst))
```

Comment:
each feature is necessarily a single value? could it include 3 values like spacing which includes value per axis?

Reply:
Yes, single value features